draft-specifications/Model.md

```diff
@@ -0,0 +1,549 @@
+> - **Python**: `''`
+>
+> †Strings participate in the [JSON subset](#json-invariants) of OCapN except
+> any strings that contain unpaired/lone surrogates.
+>
+> A string's content must be expressible in UTF-8.
+> Some two-byte Unicode encodings, as in JavaScript strings, can contain
+> unpaired/lone surrogates that have no valid expression in any UTF and so
+> cannot be carried by OCapN.
+>
+> Notes: [January 2024 meeting
+> notes](https://github.com/ocapn/ocapn/blob/main/meeting-minutes/2024-01-09.md)
+> record that we agreed that strings can only be well-formed Unicode, i.e.,
+> cannot contain unpaired surrogates.
+> For JavaScript, if a string does not pass [the `isWellFormed`
+> predicate](https://github.com/tc39/proposal-is-usv-string), then it is not a
+> Passable string.
+
+For purposes of [Pass Invariant Equality](#pass-invariant-equality), a pair of
+Strings are equal if they have the same quantity of Unicode code points and
+have the same respective Unicode code points in order.
+
+## ByteArray
+
+An array of bytes.
+
+> - **Guile**: `#vu8()`
+> - **JavaScript**: `new ArrayBuffer()`
+> - **Python**: `b''`
+>
+> Tracking: https://github.com/ocapn/ocapn/issues/48
+>
+> At the [November 14, 2023](https://github.com/ocapn/ocapn/issues/94) meeting,
+> we agreed to settle on the prefix "byte" and agreed to decide at the next
+> meeting based on a [reactji
+> poll](https://github.com/ocapn/ocapn/issues/48#issuecomment-1811097196).
+>
+> [January 2024 meeting
+> notes](https://github.com/ocapn/ocapn/blob/main/meeting-minutes/2024-01-09.md)
+> record that we agreed on ByteArray because it was the winner of the poll, and
+> we had already agreed to resolve this specific issue by poll among these
+> three choices.
+>
+> The JavaScript representation of a ByteArray is an `ArrayBuffer` which may
+> be made immutable with the proposed JavaScript [Immutable
+> ArrayBuffer](https://github.com/tc39/proposal-immutable-arraybuffer) feature.
+
+For purposes of [Pass Invariant Equality](#pass-invariant-equality), a pair of
+ByteArrays are equal if they have the same quantity of bytes and have the same
+respective bytes in order.
+
+## Selector
+
+A string of Unicode code points excluding lone (or unpaired) surrogates
+(U+D800-U+DFFF).
+Selectors are distinguished from [String](#string)s by type, not content.
+
+> - **Guile**: symbols `'name`
+> - **JavaScript**: an object with two own properties:
+>   for the registered symbol key `passStyle`, the value is the string
+>   `selector`; and
+>   for the well-known symbol `toStringTag`, the value is the selector.
+>   ```js
+>   ({
+>     [Symbol.for('passStyle'): 'selector',
+>     [Symbol.toStringTag]: 'name',
+>   })
+>   ```
+> - **Python**: `Selector('name')` where `Selector` is imported from `ocapn`.
+>
+> A selector's content must be expressible in UTF-8.
+> Strings in UTF-16 can express unpaired/lone surrogates that cannot be
+> expressed in any UTF and so cannot be carried by OCapN.
```

Comment:
UTF-16 does not allow unpaired surrogates, but the arbitrary 16-bit UCS-2/"WTF-16" encoding used in e.g. JavaScript does.

```suggestion
> Some two-byte Unicode encodings, as in JavaScript strings, can contain
> unpaired/lone surrogates that have no valid expression in any UTF and so
> cannot be carried by OCapN.
```